### ISSUE TITLE
[Components] In some rare cases the method component.stop() can work incorrectly

### ISSUE THREAD

**Issue by drew2a** [Tribler/tribler#6358]
During the refactoring (#6335), we found the following potential problem:

In some rare cases the method [component.stop()](https://github.com/Tribler/tribler/blob/main/src/tribler-core/tribler_core/components/base.py#L165) can work incorrectly.

The current implementation (simplified) is the following:

```python
class Session:
    async def start(self, failfast=True):
        coros = [comp.start() for comp in self.components.values()]
        await gather(*coros, return_exceptions=not failfast)

    async def shutdown(self):
        await gather(*[create_task(component.stop()) for component in self.components.values()])
```

That means whatever `comp.start()` was failed or succeeded, the corresponding `component.stop()` will be called during a shutdown.

That means `component.stop()` should distinguish initialized and uninitialized fields of the corresponding class (this is not currently being done).

**Comment by Solomon1732** [comment]
What is the way to check for a successfully started component? I'm guessing `comp.started.is_set()`?

**Comment by kozlovsky** [comment]
@Solomon1732 yes, but in 2730ab8f it was renamed to `comp.started_event.is_set()` to avoid possible mistake with a Boolean flag:

```python
if comp.started:   # will evaluate to True even if component do not started yet 
    ...
```

Now it should look clearer:
```python
if comp.started_event.is_set():
    ...
```
